src/docfx/lib/HttpClientUtility.cs

```diff
@@ -39,21 +39,21 @@ public static async Task<HttpResponseMessage> GetAsync(string requestUri, HttpSe
             return null;
         }
 
-        public static async Task<HttpResponseMessage> PutAsync(string requestUri, HttpContent content, HttpSecretConfig[] rules)
+        public static async Task<HttpResponseMessage> PutAsync(string requestUri, HttpContent content, Config config)
         {
-            var message = CreateHttpRequestMessage(requestUri, rules);
+            var message = CreateHttpRequestMessage(requestUri, config);
             message.Method = HttpMethod.Put;
             message.Content = content;
             return await s_httpClient.SendAsync(message);
         }
 
-        private static HttpRequestMessage CreateHttpRequestMessage(string requestUri, HttpSecretConfig[] rules)
+        private static HttpRequestMessage CreateHttpRequestMessage(string requestUri, Config config)
         {
             var message = new HttpRequestMessage();
 
-            foreach (var rule in rules)
+            foreach (var (baseUrl, rule) in config.Http)
             {
-                if (rule.Match(requestUri))
+                if (requestUri.StartsWith(baseUrl))
```

Comment:
>StartsWith [](start = 31, length = 10)

case sensitive?

Reply:
yes. Url comparison is complex, it is temporarily using case sensitive starts with.